geth/api/backend.go

```diff
@@ -25,56 +26,51 @@ const (
 // StatusBackend implements Status.im service
 type StatusBackend struct {
 	mu              sync.Mutex
-	nodeManager     *node.NodeManager
-	accountManager  common.AccountManager
-	txQueueManager  *transactions.Manager
-	jailManager     jail.Manager
-	newNotification common.NotificationConstructor
 	connectionState ConnectionState
+	Provider        *provider.ServiceProvider
```

Comment:
I am not really sure this change here is needed. `StatusBackend` is used as an entry object and I don't see any advantage in having a `Provider` here.